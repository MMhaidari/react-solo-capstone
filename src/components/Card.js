/* eslint-disable react/prop-types */
import React from 'react';

function Card({
  color, name, population, flag,
}) {
  return (
    <div className={`homeCard ${color[0]}`}>
      <div className="homeCardTop">
        <i className="fa-regular fa-circle-right cardIcon" />
        <img className="homeCardTopImg" src={flag} alt={name} />
      </div>
      <div className="homeCardBottom">
        <p className="homeCardTitle">{name}</p>
        <p className="homeCardNumber">{`${population} person`}</p>
      </div>
    </div>
  );
}

export default Card;
