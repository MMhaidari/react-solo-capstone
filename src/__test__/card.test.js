import React from 'react';
import renderer from 'react-test-renderer';
import Card from '../components/Card';

describe('Card', () => {
  it('renders with red color', () => {
    const tree = renderer.create(
      <Card
        color="red"
        name="Country Name"
        population={1000000}
        flag="https://example.com/flag.png"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with blue color', () => {
    const tree = renderer.create(
      <Card
        color="blue"
        name="Country Name"
        population={1000000}
        flag="https://example.com/flag.png"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with long country name', () => {
    const tree = renderer.create(
      <Card
        color="green"
        name="This is a very long country name that should wrap to multiple lines"
        population={1000000}
        flag="https://example.com/flag.png"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with a large population', () => {
    const tree = renderer.create(
      <Card
        color="yellow"
        name="Country Name"
        population={1000000000}
        flag="https://example.com/flag.png"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with a different flag image', () => {
    const tree = renderer.create(
      <Card
        color="orange"
        name="Country Name"
        population={1000000}
        flag="https://example.com/different-flag.png"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
