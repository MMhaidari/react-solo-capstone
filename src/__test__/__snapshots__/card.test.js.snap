// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders with a different flag image 1`] = `
<div
  className="homeCard o"
>
  <div
    className="homeCardTop"
  >
    <i
      className="fa-regular fa-circle-right cardIcon"
    />
    <img
      alt="Country Name"
      className="homeCardTopImg"
      src="https://example.com/different-flag.png"
    />
  </div>
  <div
    className="homeCardBottom"
  >
    <p
      className="homeCardTitle"
    >
      Country Name
    </p>
    <p
      className="homeCardNumber"
    >
      1
      M
    </p>
  </div>
</div>
`;

exports[`Card renders with a large population 1`] = `
<div
  className="homeCard y"
>
  <div
    className="homeCardTop"
  >
    <i
      className="fa-regular fa-circle-right cardIcon"
    />
    <img
      alt="Country Name"
      className="homeCardTopImg"
      src="https://example.com/flag.png"
    />
  </div>
  <div
    className="homeCardBottom"
  >
    <p
      className="homeCardTitle"
    >
      Country Name
    </p>
    <p
      className="homeCardNumber"
    >
      1000
      M
    </p>
  </div>
</div>
`;

exports[`Card renders with blue color 1`] = `
<div
  className="homeCard b"
>
  <div
    className="homeCardTop"
  >
    <i
      className="fa-regular fa-circle-right cardIcon"
    />
    <img
      alt="Country Name"
      className="homeCardTopImg"
      src="https://example.com/flag.png"
    />
  </div>
  <div
    className="homeCardBottom"
  >
    <p
      className="homeCardTitle"
    >
      Country Name
    </p>
    <p
      className="homeCardNumber"
    >
      1
      M
    </p>
  </div>
</div>
`;

exports[`Card renders with long country name 1`] = `
<div
  className="homeCard g"
>
  <div
    className="homeCardTop"
  >
    <i
      className="fa-regular fa-circle-right cardIcon"
    />
    <img
      alt="This is a very long country name that should wrap to multiple lines"
      className="homeCardTopImg"
      src="https://example.com/flag.png"
    />
  </div>
  <div
    className="homeCardBottom"
  >
    <p
      className="homeCardTitle"
    >
      This is a very long country name that should wrap to multiple lines
    </p>
    <p
      className="homeCardNumber"
    >
      1
      M
    </p>
  </div>
</div>
`;

exports[`Card renders with red color 1`] = `
<div
  className="homeCard r"
>
  <div
    className="homeCardTop"
  >
    <i
      className="fa-regular fa-circle-right cardIcon"
    />
    <img
      alt="Country Name"
      className="homeCardTopImg"
      src="https://example.com/flag.png"
    />
  </div>
  <div
    className="homeCardBottom"
  >
    <p
      className="homeCardTitle"
    >
      Country Name
    </p>
    <p
      className="homeCardNumber"
    >
      1
      M
    </p>
  </div>
</div>
`;
